src/Simple.OData.Client.V4.Adapter/CommandFormatter.cs

```diff
@@ -106,19 +98,52 @@ protected override void FormatInlineCount(IList<string> commandClauses)
+                if (expandsToCollection && !ReferenceEquals(association.FilterExpression, null))
+                {
+                    var associatedEntityCollection = _session.Metadata.GetEntityCollection(
+                        _session.Metadata.GetNavigationPropertyPartnerTypeName(entityCollection.Name, associationName));
+                    clauses.Add(
+                        $"{ODataLiteral.Filter}={EscapeUnescapedString(association.FilterExpression.Format(new ExpressionContext(_session, associatedEntityCollection, null, command.DynamicPropertiesContainerName)))}");
+                }
 
-            if (expandsToCollection && orderbyColumns.Any())
-            {
-                var columns = string.Join(",", SelectPathSegmentColumns(orderbyColumns, null, entityCollection)
-                    .Select(x => x.Key + (x.Value ? " desc" : string.Empty)).ToList());
-                if (!string.IsNullOrEmpty(columns))
-                    clauses.Add($"{ODataLiteral.OrderBy}={columns}");
+                if (association.ExpandAssociations.Any())
+                {
+                    var associatedEntityCollection = _session.Metadata.GetEntityCollection(
+                        _session.Metadata.GetNavigationPropertyPartnerTypeName(entityCollection.Name, associationName));
+                    var expandAll = association.ExpandAssociations.FirstOrDefault(a => a.Name == StarString);
+                    if (expandAll != null)
+                    {
+                        clauses.Add($"{ODataLiteral.Expand}=*");
+                    }
+                    else
+                    {
+                        clauses.AddRange(association.ExpandAssociations
+                            .Where(
+                                a => _session.Metadata.HasNavigationProperty(associatedEntityCollection.Name, a.Name))
+                            .Select(a =>
+                                FormatExpansionSegment(a, associatedEntityCollection, ODataExpandOptions.ByValue(),
+                                    command,
+                                    false))
+                            .Select(formattedExpand => $"{ODataLiteral.Expand}={formattedExpand}"));
+                    }
+
+                    var selectColumns = string.Join(",", association.ExpandAssociations.ToList()
+                        .Where(a => a.Name != StarString &&
+                                    !_session.Metadata.HasNavigationProperty(associatedEntityCollection.Name, a.Name))
+                        .Select(a => a.Name));
+                    if (!string.IsNullOrEmpty(selectColumns))
+                        clauses.Add($"{ODataLiteral.Select}={selectColumns}");
+                }
+
+                if (expandsToCollection && association.OrderByColumns.Any())
+                {
+                    var columns = string.Join(",", association.OrderByColumns
+                        .Select(o => o.Name + (o.Descending ? " desc" : string.Empty)).ToList());
```

Comment:
Is the `ToList` required here?

Reply:
You're right, the .ToList() call is not necessary here. I will remove this call.